### PR TITLE
Symbolic expressions

### PR DESCRIPTION
**Pull request by tobiasduerschmid**
Add symbolic expressions with logical and math operators

### CODE REVIEW

File: include/rosdiscover-clang/ApiCall/Calls/Util.h [hunk 1]
```diff
@@ -87,8 +87,8 @@ namespace api_call {
 clang::APValue const * evaluateNumber(
   const std::string debugTag, 
   const clang::Expr *expr,
-  const clang::ASTContext &Ctx
-) {
+  const clang::ASTContext &Ctx,
+  bool debugPrint=true) {
```
Comment (ChrisTimperley) [inline include/rosdiscover-clang/ApiCall/Calls/Util.h:91]: nitpick: put `) {` on a separate line for consistency. (we should really figure out how to get clang-format running via GitHub Actions!)

Comment (tobiasduerschmid) [reply include/rosdiscover-clang/ApiCall/Calls/Util.h:91]: done

File: include/rosdiscover-clang/Ast/Stmt/EnumReference.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#include "../../Value/Value.h"
+
+namespace rosdiscover {
+
+class SymbolicEnumReference : public SymbolicDeclRef {
+public:
+  SymbolicEnumReference(
+    std::string typeName,
+    std::string name,
+    long value
```
[the remark this answers is not in the document]
Comment (tobiasduerschmid) [reply include/rosdiscover-clang/Ast/Stmt/EnumReference.h:17]: `enum class` element's don't have corresponding int values? Interesting. I guess I'll need to look into this in a separate.

File: include/rosdiscover-clang/Ast/Stmt/VariableReference.h [hunk 3]
```diff
@@ -10,6 +10,20 @@ namespace rosdiscover {
...
+    std::string typeName,
+    std::string name,
+    bool isFileVarDecl,
+    bool isLocalVarDeclOrParm,
+    bool isModulePrivate
+  ) : SymbolicDeclRef(isInstanceMember, isClassMember, typeName, name),
+      isFileVarDecl(isFileVarDecl),
+      isLocalVarDeclOrParm(isLocalVarDeclOrParm),
+      isModulePrivate(isModulePrivate)
+  {}
```
Comment (ChrisTimperley) [inline include/rosdiscover-clang/Ast/Stmt/VariableReference.h:25]: this is much easier to read. thanks!

**Pull request merged**